### PR TITLE
Refactor contracts directory and add ERC721GatedPaymaster with unit test

### PR DESCRIPTION
**Pull request by dutterbutter**
**Changes introduced:**

* Reorganized contract directory structure for better clarity
* Added `ERC721GatedPaymaster` contract
* Included unit tests for `ERC721GatedPaymaster` contract
* Removed commented out examples
* Updated and expanded `README.md` with new contract details
* Implemented code formatting standardization with 'yarn format' command

### CODE REVIEW

File: contracts/README.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# Paymaster Examples 📁
+
+Welcome to the world of Paymasters! 🎉
+
+This directory contains several example Smart Contracts and a utility. These Smart Contracts are purely illustrative, showcasing different ways that a Paymaster Smart Contract can be structured. ⚠️ They are **not** designed for production use.
```
Comment (uF4No) [inline contracts/README.md:5]:
```suggestion
This directory contains several example Smart Contracts and a utility. These Smart Contracts are purely illustrative, showcasing different ways that a Paymaster Smart Contract can be structured. ⚠️ These contracts have **not been audited** and  are **not** designed for production use.  Please use these just as a reference.
```

**Comment by uF4No** [comment]
can we rename the `paymaster` folder to `paymasters`.

Also, I did not include a test for the `withdraw` method, can add it? It should be the same for all paymasters

**Comment by dutterbutter** [comment]
@uF4No made requested changes. The next PR won't be so large.

**Comment by uF4No** [comment]
I just noticed that the `erc20gated` paymaster is actually a paymaster that accepts a fixed amount of tokens in exchange for the gas fees.
My idea for `erc20gated` is that it will pay the gas fees if the user has a balance > 0. Normally this is useful for DAOs that have a token for governance so, if someone holds any amount of that token, the paymaster will pay the gas fees.
I think for now we can just rename `erc20gated` to `erc20fixed` as the current functionality is also valid

**Review by uF4No — APPROVED** [review]

**Pull request merged**